configs/components/_base-ruby.rb

```diff
@@ -50,15 +50,20 @@
   pkg.environment 'optflags', settings[:cflags] + ' -O3'
 elsif platform.is_macos?
   pkg.environment 'optflags', settings[:cflags]
-  if platform.is_cross_compiled?
+  if platform.architecture == 'arm64'
     # Pin to an older version of ruby@2.5. This can be removed once we're no longer cross-compiling
     if ruby_version_y == "2.5"
       pkg.build_requires "puppetlabs/puppet/ruby@2.5"
     else
       pkg.build_requires "ruby@#{ruby_version_y}"
     end
-    pkg.environment 'CC', 'clang -target arm64-apple-macos11' if platform.name =~ /osx-11/
-    pkg.environment 'CC', 'clang -target arm64-apple-macos12' if platform.name =~ /osx-12/
+    
+    if platform.is_cross_compiled?
+      pkg.environment 'CC', 'clang -target arm64-apple-macos11' if platform.name =~ /osx-11/
+      pkg.environment 'CC', 'clang -target arm64-apple-macos12' if platform.name =~ /osx-12/
+    else
+      pkg.environment 'CC', 'clang'
+    end
```

Comment:
Adding a note that we talked about this and discussed how we should continue using `is_cross_compiled?` And within the body, we need to express our build dependency on a prebuilt version of ruby and we need to specify the build `target`:
 
```ruby
if platform.is_cross_compiled?
  # Pin to an older version of ruby@2.5. This can be removed once we're no longer cross-compiling
  if ruby_version_y == "2.5"
    pkg.build_requires "puppetlabs/puppet/ruby@2.5"
  else
    pkg.build_requires "ruby@#{ruby_version_y}"
  end
  pkg.environment 'CC', 'clang -target arm64-apple-macos11' if platform.name =~ /osx-11/
  pkg.environment 'CC', 'clang -target arm64-apple-macos12' if platform.name =~ /osx-12/
else
  pkg.environment 'CC', 'clang'
end
```